cmem/encoder_test.go

```diff
@@ -0,0 +1,97 @@
+package cmem
+
+import (
+	"encoding/binary"
+	"reflect"
+	"testing"
+)
+
+type testStruct struct {
+	Val1 uint8
+	Val2 uint16
+	Val3 uint32
+	Val4 uint64
+}
+
+func testCmemStructAlignment(t *testing.T, endianess binary.ByteOrder) {
```

Comment:
I was thinking more along the lines of:
```go
func TestEncode(t *testing.T) {
	for i, tc := range []struct {
		v    interface{}
		want []byte
	}{
		{
			v: struct {
				V1 uint8
				V2 uint64
				V3 uint8
			}{1, 2, 3},
			want: []byte("some binary string"),
		},
		{
			v: []int32{1, 20, 300, 4000, 50000, 600000, 7000000, 8000000},
			want: []byte("some other binary string"),
		},
	} {
		t.Run(strconv.Itoa(i), func(t *testing.T) {
			oldEndian := nativeEndian
			nativeEndian = binary.LittleEndian
			defer func() { oldEndian = nativeEndian }()
			var enc Encoder
			err := enc.Encode(tc.v)
			if err != nil {
				t.Fatalf("could not encode: %v", err)
			}
			if !bytes.Equal(enc.Buf, tc.want) {
				t.Fatalf("encoding error:\ngot = %v\nwant= %v", enc.Buf, tc.want)
			}
		})
	}
}

```

Reply:
Quite beautiful, let me work with that :)